### ISSUE TITLE
Font-lock block literal spilling out

### ISSUE THREAD

**Issue by dogwynn** [yoshiki/yaml-mode#60]
#### Problem

When leaving a block literal to continue a dictionary definition, the rest of the dictionary is font-locked as a string.

![screen shot 2018-02-27 at 3 49 01 pm](https://user-images.githubusercontent.com/1621000/36757090-48c3e280-1bd6-11e8-8a67-0cb272ccb56a.png)

This implies that the rest of the dictionary is part of the string literal, which is not true.

```
In [5]: yaml.load('''
   ...: - text: >-
   ...:     This is a block literal with font-lock spilling out
   ...:   a: 1 
   ...:   b: 2 
   ...:   c: {} 
   ...: ''')
Out[5]: 
[{'a': 1,
  'b': 2,
  'c': {},
  'text': 'This is a block literal with font-lock spilling out'}]
```

This is new behavior. 

#### Current Solutions

The addition of comment symbols seems to "fix" the issue.

![screen shot 2018-02-27 at 3 57 39 pm](https://user-images.githubusercontent.com/1621000/36757365-fd5e20a2-1bd6-11e8-826d-0e4ff08e97d5.png)

However, when the comment symbol is deleted, the font-lock spills back in:

![screen shot 2018-02-27 at 4 06 57 pm](https://user-images.githubusercontent.com/1621000/36757880-47e0ca3e-1bd8-11e8-966e-4794e8f01fe5.png)

**Comment by dogwynn** [comment]
[v0.0.11](https://github.com/yoshiki/yaml-mode/tree/v0.0.11) does not have this problem.

**Comment by wasamasa** [comment]
Well this is funny. The block literal code existed for quite some time in the font locking rules, but wasn't applied properly, so I fixed that in my recent commits.  Apparently this reveals that its logic isn't perfect.  Your PR attacks the right place, but you'd need to understand the existing logic to make progress on this problem.

**Comment by dogwynn** [comment]
Yeah, I spent a bit of time last night dowsing the variable space to try to wrap my brain around the logic to no avail. I hate that this isn't working. I'm using YAML to persist my exams and quizzes, so I make extensive use of block literals. 

For the time being, I've reverted to v0.0.11 until I can get some time to grok the font-locking code.

Thanks for the module all the same!
